device/esp_tinyusb/include/tinyusb_msc.h

```diff
@@ -0,0 +1,286 @@
+                                             *   - Called before and after switching access to the storage between USB and App.
+                                             *   - Called when a mount or unmount operation has not been completed.
+                                             */
+
+    void *callback_arg;                     /*!< Argument passed to the user callback.
+                                             *   - Can be used to pass user context or additional data.
+                                             */
+} tinyusb_msc_driver_config_t;
+
+// ----------------------------- Driver API ---------------------------------
+
+/**
+ * @brief Install TinyUSB MSC driver
+ *
+ * This function initializes the TinyUSB MSC driver with the provided configuration.
+ *
+ * @param[in] config Pointer to the configuration structure for TinyUSB MSC driver
+ * @return
+ *   - ESP_OK: Installation successful
+ *   - ESP_ERR_INVALID_STATE: Driver is already installed
+ *   - ESP_ERR_INVALID_ARG: Invalid input argument
+ *   - ESP_ERR_NO_MEM: Not enough memory to install the driver
+ */
+esp_err_t tinyusb_msc_driver_install(const tinyusb_msc_driver_config_t *config);
+
+/**
+ * @brief Uninstall TinyUSB MSC driver
+ *
+ * This function deinitializes the TinyUSB MSC driver and releases any resources allocated during initialization.
+ *
+ * @return
+ *    - ESP_OK: Uninstallation successful
+ *    - ESP_ERR_NOT_FOUND: Driver is not installed
+ *    - ESP_ERR_INVALID_STATE: Driver is in incorrect state: storage is still mounted
+ */
+esp_err_t tinyusb_msc_driver_uninstall(void);
+
+/**
+ * @brief Initialize TinyUSB MSC SPI Flash storage
+ *
+ * This function initializes the TinyUSB MSC storage interface with the provided configuration.
+ *
+ * @param[in] config Pointer to the configuration structure for TinyUSB MSC storage
+ * @param[out] handle Pointer to the storage handle
+ *
+ * @return
+ *    - ESP_OK: Initialization successful
+ *    - ESP_ERR_INVALID_ARG: Invalid input argument
+ *    - ESP_ERR_NOT_SUPPORTED: TinyUSB buffer size is less than the Wear Levelling sector size
+ *    - ESP_ERR_NO_MEM: Not enough memory to initialize storage
+ */
+esp_err_t tinyusb_msc_storage_new_spiflash(const tinyusb_msc_storage_config_t *config, tinyusb_msc_storage_handle_t *handle);
+
+#if (SOC_SDMMC_HOST_SUPPORTED)
+/**
+ * @brief Initialize TinyUSB MSC storage with SDMMC or EMMC
+ *
+ * This function initializes the TinyUSB MSC storage interface with MMC as the storage medium.
+ *
+ * @param[in] config Pointer to the configuration structure for TinyUSB MSC storage with SDMMC or EMMC
+ * @param[out] handle Pointer to the storage handle
+ *
+ * @return
+ *    - ESP_OK: Initialization successful
+ *    - ESP_ERR_INVALID_ARG: Invalid input argument
+ *    - ESP_ERR_NO_MEM: Not enough memory to initialize storage
+ */
+esp_err_t tinyusb_msc_storage_new_mmc(const tinyusb_msc_storage_config_t *config, tinyusb_msc_storage_handle_t *handle);
```

Comment:
Correct spelling is SD/MMC.

So, uSD is a part of SD and eMMC is a part of MMC. 

So, previous naming was correct. 

```suggestion
 * @brief Initialize TinyUSB MSC SD/MMC storage
 *
 * This function initializes the TinyUSB MSC storage interface with SD/MMC as the storage medium.
 *
 * @param[in] config Pointer to the configuration structure for TinyUSB MSC storage with SD/MMC
 * @param[out] handle Pointer to the storage handle
 *
 * @return
 *    - ESP_OK: Initialization successful
 *    - ESP_ERR_INVALID_ARG: Invalid input argument
 *    - ESP_ERR_NO_MEM: Not enough memory to initialize storage
 */
esp_err_t tinyusb_msc_storage_new_sdmmc(const tinyusb_msc_storage_config_t *config, tinyusb_msc_storage_handle_t *handle);
```